### PR TITLE
fix: `view_item_list` event being fired without products

### PR DESCRIPTION
**Pull request by filipewl**
## What's the purpose of this pull request?

Fixed an issue where the `view_item_list` analytics event is being fired with an empty list of `items` when viewing the list of products on the PLPs. We also configured the `ProductShelf` and `ProductTiles` components to fire this event because they render a list of products (e.g. "Most Wanted" and "Just Arrived now!" sections on the Home page).

## How to test it?

Integration tests were added, and they assert the event is fired as expected.

You can check out the `dataLayer` events being fired on https://preview-242--nextjs.preview.vtex.app/ following this guide [See events pushed to `dataLayer`](https://www.faststore.dev/how-to-guides/troubleshooting/analytics-and-partytown#see-events-pushed-to-datalayer):

Before|After
-|-
CleanShot 2022-09-13 at 17 37 03|CleanShot 2022-09-13 at 17 36 06
CleanShot 2022-09-13 at 17 33 52|CleanShot 2022-09-13 at 17 33 29

## References

- [[Internal] Jira task FS-469](https://vtex-dev.atlassian.net/browse/FS-469)
- https://www.faststore.dev/how-to-guides/troubleshooting/analytics-and-partytown#see-events-pushed-to-datalayer for explaining how to check events being pushed to the `dataLayer`.

### CODE REVIEW

File: cypress/integration/analytics.test.js [hunk 1]
```diff
@@ -268,10 +268,56 @@ describe('view_item_list event', () => {
...
+          const event = window.dataLayer.find(
+            (evt) => evt.event === 'view_item_list'
+          )
+
+          expect(event.ecommerce.items.length).to.eq(5)
+        })
+      })
+  })
+
+  it('is sent when viewing a product tiles', () => {
```
Comment (filipewl) [inline cypress/integration/analytics.test.js:303]: Before the fix viewing these components would not trigger the `view_item_list` event:
CleanShot 2022-09-13 at 17 18 14

File: src/components/sections/Section/Section.tsx [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
 
-function Section({ className = '', ...otherProps }: Props) {
-  return <section className={`section ${className}`} {...otherProps} />
-}
+const Section = forwardRef<HTMLDivElement, Props>(function Section(
+  { className = '', ...otherProps },
+  ref
+) {
+  return (
+    <section ref={ref} className={`section ${className}`} {...otherProps} />
```
Comment (filipewl) [inline src/components/sections/Section/Section.tsx:11]: Needed to allow forwarding a `ref` for the in-view observer.

File: src/components/sections/ProductTiles/ProductTiles.tsx [hunk 3]
```diff
@@ -7,7 +11,7 @@ import type { ProductsQueryQueryVariables } from '@generated/graphql'
 import Section from '../Section'
 
 interface ProductTilesProps extends Partial<ProductsQueryQueryVariables> {
-  title: string | JSX.Element
```
Comment (filipewl) [inline src/components/sections/ProductTiles/ProductTiles.tsx:10]: We need the section/item list's name as a string to use in [the field `item_list_name` for the `view_item_list` event](https://developers.google.com/analytics/devguides/collection/ga4/reference/events#parameters_35) so allowing a `JSX.Element` became unusable. We don't have instances where `title` is used with this type, so removing it seems fine.

**Review by icazevedo — APPROVED** [review]
Awesome! Great work 👏

**Pull request merged**